### PR TITLE
workflows: replace depracted set-env with github envs

### LINKED ISSUE #2231: GitHub Actions: The `set-env` command is deprecated and will be disabled soon.
The `set-env` command is deprecated and will be disabled soon. Please upgrade to using Environment Files. For more information see: https://github.blog/changelog/2020-10-01-github-actions-deprecating-set-env-and-add-path-commands/

As seen at https://github.com/nodejs/node-gyp/actions

* **Node Version**: 
* **Platform**: 
* **Compiler**: 
* **Module**: 

<details><summary>Verbose output (from npm or node-gyp):</summary>

```
Paste your log here, between the backticks. It can be:
  - npm --verbose output,
  - or contents of npm-debug.log,
  - or output of node-gyp rebuild --verbose.
Include the command you were trying to run.

This should look like this:

>npm --verbose
npm info it worked if it ends with ok
npm verb cli [
npm verb cli   'C:\\...\\node\\13.9.0\\x64\\node.exe',
npm verb cli   'C:\\...\\node\\13.9.0\\x64\\node_modules\\npm\\bin\\npm-cli.js',
npm verb cli   '--verbose'
npm verb cli ]
npm info using npm@6.13.7
npm info using node@v13.9.0

Usage: npm <command>
(...)
```

</details>

### PR DESCRIPTION
**Pull request by MadMango**
Fixes #2231 

##### Checklist

- [x] `npm install && npm test` passes
- [ ] tests are included 
- [ ] documentation is changed or added
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

##### Description of change

set-env is deprecated and builds will fail, this should fix it

### CODE REVIEW

**Comment by MadMango** [comment]
Just realised this is for powersheel or cmd and my unix fix won't do here, I'll try come up with something else but can someone confirm if this will run in CMD or powershell?

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -32,8 +32,8 @@ jobs:
       - name: Set Windows environment
         if: matrix.os == 'windows-latest'
         run:
```
Comment (richardlau) [inline .github/workflows/tests.yml:34]: Looks like an existing bug in the workflow, but this should be:

```suggestion
        run: |
```
to preserve the line break between the two subsequent `echo ...` lines.

**Comment by richardlau** [comment]
> Just realised this is for powersheel or cmd and my unix fix won't do here, I'll try come up with something else but can someone confirm if this will run in CMD or powershell?

Since we don't override the default I believe actions on Windows run in powershell by default: https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell

**Comment by MadMango** [comment]
Tried to help but I'm out of ideas I'm afraid, setting these variables through powershell or python doesn't seem to make a difference:

image
image

In both cases when trying to print these variables they are not defined:

image

Any other ideas?

File: gyp/pylib/gyp/MSVSVersion.py [hunk 2]
```diff
@@ -549,6 +549,9 @@ def SelectVisualStudioVersion(version="auto", allow_fallback=True):
         "2017": ("15.0",),
         "2019": ("16.0",),
     }
+    print('ENV OVERRIDES: \n')
+    print(os.environ.get("GYP_MSVS_OVERRIDE_PATH"))
+    print(os.environ.get("GYP_MSVS_VERSION"))
```
Comment (cclauss) [inline gyp/pylib/gyp/MSVSVersion.py:554]:
```suggestion
    for key in ("GYP_MSVS_OVERRIDE_PATH", "GYP_MSVS_VERSION"):
        print("{}: {}".format(key, os.environ.get(key)))
```

**Comment by gengjiawen** [comment]
I think I find the right way in https://github.com/nodejs/node-gyp/pull/2285, can you cherrypick it here ?

**Comment by MadMango** [comment]
> I think I find the right way in #2285, can you cherrypick it here ?

Well you got it working in that one, let's just close this one and go with yours :) 
I would've never quessed the `$Env:GITHUB_ENV` syntax though 🤯

**Comment by gengjiawen** [comment]
Ok， I find the other part is still useful, I will try to import to my patch.

Thanks for taking time on this :)

> I would've never quessed the $Env:GITHUB_ENV syntax though 🤯

Yeap, github documentation is useless in this case. I searched lots of code find this out.

**Pull request closed without merging**